### PR TITLE
SIP - Allow multiple implicit parameter lists

### PR DESCRIPTION
**Pull request by mgttlinger**

### CODE REVIEW

File: sips/pending/_posts/2016-04-21-multiple-implicit-parameter-lists.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+now(1, implicitly)
+
+def then(implicit a: Int)(implicit b: String) = b * a
+then(1)
+{% endhighlight %}
+
+## Implementation
+
+The implementation of this change is as simple as removing the restriction of having only one parameter list in the `Parser`. A pull request (#5108) from Miles Sabin already implements this change as well as an accompanying test file. The change should not affect any existing code as every Scala statement that was valid before will still be valid after the change. There will even be binary compatibility in both directions.
+The desugaring ob context and view bounds will be added on the left of the leftmost implicit parameter list to go with the general expectation from the current implementation.
```
Comment (etorreborre) [inline sips/pending/_posts/2016-04-21-multiple-implicit-parameter-lists.md]: s/ob/of

File: sips/pending/_posts/2016-04-21-multiple-implicit-parameter-lists.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+## History
+
+| Date          | Version       |
+|---------------|---------------|
+| Apr 21st 2016 | Initial Draft |
+
+## Motivation
+
+Currently, the implicit parameters of a function are limited to being placed in a single, final, parameter list of a function. This prevents later parameters from using dependant types of former ones, as this is only allowed between, but not inside a single parameter list.
```
Comment (etorreborre) [inline sips/pending/_posts/2016-04-21-multiple-implicit-parameter-lists.md]: s/dependant/dependent

**Comment by retronym** [comment]
@travisbrown wrote a [good post](https://meta.plasm.us/posts/2015/07/11/roll-your-own-scala/) showing a motivating use case and a workaround, this might be a good reference to add to this SIP.

**Comment by jvican** [comment]
Hello @mgttlinger and @milessabin!
  
This proposal will be reviewed by the Committee shortly, according to [the new process](http://docs.scala-lang.org/sips/sip-submission.html). Thanks for submitting it!

**Comment by milessabin** [comment]
We are reevaluating the this proposal and we want to make changes based on further implementation and experience in use. I suggest that this proposal be withdrawn from review for the time being. We will come back to it when we are ready.

**Comment by jvican** [comment]
Thanks for the status update! This means that we will not start the review process for this proposal any time soon either. I'm closing the PR, please feel free to reopen and update the proposal when you feel ready.

**Pull request closed without merging**